signer/src/transaction_signer.rs

```diff
@@ -652,21 +628,28 @@ where
                 }
 
                 let db = self.context.get_storage();
-                Self::validate_bitcoin_sign_request(&db, &request.message).await?;
-                self.relay_message(msg.txid, &msg.inner, bitcoin_chain_tip)
-                    .await?;
+                let sighash = Self::validate_bitcoin_sign_request(&db, &request.message).await?;
+                let id = sighash.into();
+                let response = self
+                    .relay_message(id, msg.txid, &msg.inner, bitcoin_chain_tip)
+                    .await;
+
+                self.wsts_state_machines.pop(&id);
+                response?;
```

Comment:
Are we sure we can safely pop here? eg, if `relay_message` failed something, will the wsts coordinator try again? If so, will it do it from scratch or we need to have the state machine still ready?

Reply:
I didn't think that the WSTS coordinator retried anything, but let me check.

Reply:
hmmm, I think my above message is better tailored to the question here. Basically, I think that we can safely pop here. I don't think that our use of WSTS tries again but if it did we should require new nonces (I think that's the safer thing to do). So I'm okay with deleting the state machine and requiring the creation of a new one with fresh nonces.